probe-rs/src/memory/mod.rs

```diff
@@ -26,6 +26,37 @@ pub trait MemoryInterface {
     /// Read a block of 8bit words at `address`.
     fn read_8(&mut self, address: u32, data: &mut [u8]) -> Result<(), error::Error>;
 
+    /// Reads bytes using 32 bit memory access. Address must be 32 bit aligned
+    /// and data must be an exact multiple of 4.
+    fn read_mem_32bit(&mut self, address: u32, data: &mut [u8]) -> Result<(), error::Error> {
+        // Default implementation uses `read_32`, then converts u32 values back
+        // to bytes. Assumes target is little endian. May be overridden to
+        // provide an implementation that avoids heap allocation and endian
+        // conversions. Must be overridden for big endian targets.
+        let mut buffer = vec![0u32; data.len() / 4];
```

Comment:
Please add a check here to ensure that `data.len()` is actually a multiple of 4. Otherwise someone might read less bytes than expected.

Reply:
Done